### DISCUSSION TITLE
Error #403 with v 0.7.2

### DISCUSSION THREAD

**Discussion by marcnshapiro** [alvarobartt/investiny#59] [Q&A] [answered]
Installed Investiny 0.6.1, made a few changes to my program and everything was working again.  This morning I upgraded to 0.7.2 and I get no data.  Tested it on the console and verified that I am getting the dreaded Error #403

**Comment by alvarobartt** [comment, marked as answer]
Hi again @marcnshapiro!

So I've just tested `investiny` and it seems to be working fine again... I assume their Cloudflare has some limitations but it's not blacklisting every IP forever, just after a certain number of requests...

Look:

```python
(investiny-py3.9) alvarobartt@Alvaros-MacBook-Air investiny % poetry run python
Python 3.9.6 (default, Aug  5 2022, 15:21:02) 
[Clang 14.0.0 (clang-1400.0.29.102)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from investiny import historical_data
>>> d = historical_data(investing_id=6408)
>>> d
{'date': ['09/26/2022', '09/27/2022', '09/28/2022', '09/29/2022', '09/30/2022', '10/03/2022', '10/04/2022', '10/05/2022', '10/06/2022', '10/07/2022', '10/10/2022', '10/11/2022', '10/12/2022', '10/13/2022', '10/14/2022', '10/17/2022', '10/18/2022', '10/19/2022', '10/20/2022', '10/21/2022'], 'open': [149.66000366211, 152.74000549316, 147.63999938965, 146.10000610352, 141.2799987793, 138.21000671387, 145.0299987793, 144.07499694824, 145.80999755859, 142.53999328613, 140.41999816895, 139.89999389648, 139.13000488281, 134.99000549316, 144.30999755859, 141.06500244141, 145.49000549316, 141.69000244141, 143.02000427246, 142.96000671387], 'high': [153.7700958252, 154.7200012207, 150.64140319824, 146.7200012207, 143.10000610352, 143.07000732422, 146.2200012207, 147.38000488281, 147.53999328613, 143.10000610352, 141.88999938965, 141.35000610352, 140.36000061035, 143.58999633789, 144.52000427246, 142.89999389648, 146.69999694824, 144.94920349121, 145.88999938965, 147.83999633789], 'low': [149.63999938965, 149.94500732422, 144.83999633789, 140.67999267578, 138, 137.68499755859, 144.25999450684, 143.00999450684, 145.2200012207, 139.44500732422, 138.57290649414, 138.2200012207, 138.16000366211, 134.36999511719, 138.19000244141, 140.27000427246, 140.61000061035, 141.5, 142.64999389648, 142.67999267578], 'close': [150.77000427246, 151.75999450684, 149.83999633789, 142.47999572754, 138.19999694824, 142.44999694824, 146.10000610352, 146.39999389648, 145.42999267578, 140.08999633789, 140.41999816895, 138.97999572754, 138.33999633789, 142.99000549316, 138.38000488281, 142.41000366211, 143.75, 143.86000061035, 143.38999938965, 147.27000427246], 'volume': [93339000, 84443000, 146691008, 128138000, 124925000, 114312000, 87134000, 79148000, 68402000, 85926000, 74591000, 77034000, 69833000, 112876000, 88237000, 84684000, 98716000, 61758000, 64277000, 85641896]}
```

**Reply by alvarobartt** [reply]
Do you mind checking now? And in case it doesn't work, can you provide me more information such as your Python version, `investiny` version, and timezone maybe? Any information is helpful in exploring this issue further. Thanks 👍🏻

**Comment by Rogerup** [comment]
Same problem here: " Request to Investing.com API failed with error code: 403"

call:  search_assets(query="Bursatil", limit=1, type="Index")

error: 
 r = httpx.get(url, params=params, headers=headers)
     35     if r.status_code != 200:
---> 36         raise ConnectionError(
     37             f"Request to Investing.com API failed with error code: {r.status_code}."
     38         )

python = 3.9.13
investiny = 0.7.2
Windows 10 Home
Timezone: Brazil, UTC-3

**Comment by maykon17** [comment]
I got the same problem in historical_data function. @alvarobartt could you help us?

Python version: '3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)]'

**Comment by elricStormBringer** [comment]
hi @alvarobartt ,I just got to know of your great project on investing.com from Stackoverflow.

Unfortunately, I am getting the dreaded 403 error messages, even though I have tried all of the solutons (except for the APIscraperLink).

Can I check if investiny has been updated/corrected for the errors? or this bug is something us (noob programmer) users still have to deal with?
